plotly/tests/test_core/test_tools/test_file_tools.py

```diff
@@ -1,31 +1,80 @@
-import plotly.tools as tls
-
-
-def test_reset_config_file():
-    tls.reset_config_file()
-    config = tls.get_config_file()
-    assert config['plotly_domain'] == 'https://plot.ly'
-    assert config['plotly_streaming_domain'] == 'stream.plot.ly'
-
-
-def test_set_config_file():
-    pd, ps = 'this', 'thing'
-    tls.set_config_file(plotly_domain=pd, plotly_streaming_domain=ps)
-    config = tls.get_config_file()
-    assert config['plotly_domain'] == pd
-    assert config['plotly_streaming_domain'] == ps
-    tls.reset_config_file()  # else every would hate me :)
-
-
-def test_credentials_tools():
-    original_creds = tls.get_credentials_file()
-    expected = ['username', 'stream_ids', 'api_key']
-    assert all(x in original_creds for x in expected)
-    # now, if that worked, we can try this!
-    tls.reset_credentials_file()
-    reset_creds = tls.get_credentials_file()
-    tls.set_credentials_file(**original_creds)
-    assert all(x in reset_creds for x in expected)
-    creds = tls.get_credentials_file()
-    assert all(x in creds for x in expected)
-    assert original_creds == creds
+from unittest import TestCase
+
+from plotly import tools
+from plotly import session
+
+
+class FileToolsTest(TestCase):
+
+    def test_set_config_file_all_entries(self):
+
+        # Check set_config and get_config return the same values
+
+        pd, ps, pa = 'this', 'thing', 'that'
+        psv = False
+        po = {'world_readable': False}
+        tools.set_config_file(plotly_domain=pd,
+                              plotly_streaming_domain=ps,
+                              plotly_api_domain=pa,
+                              plotly_ssl_verification=psv,
+                              plot_options=po)
+        config = tools.get_config_file()
+        self.assertEqual(config['plotly_domain'], pd)
+        self.assertEqual(config['plotly_streaming_domain'], ps)
+        self.assertEqual(config['plotly_api_domain'], pa)
+        self.assertEqual(config['plotly_ssl_verification'], psv)
+        self.assertEqual(config['plot_options'], po)
+
+    def test_set_config_file_two_entries(self):
+
+        # Check set_config and get_config given only two entries return the
+        # same values
+
+        pd, ps = 'this', 'thing'
+        tools.set_config_file(plotly_domain=pd, plotly_streaming_domain=ps)
```

Comment:
careful, this will set the config file locally for whoever is running the tests!

I'll make a parent testcase with some setup that can cache and take care of this for us. i'll ping you on the pr :)